### PR TITLE
switch from remote SauceLabs to local karma

### PR DESCRIPTION
**Pull request by jokeyrhyme**
- bit of path wrangling to get Require.js working for tests in both karma and our `grunt develop` server
- `chai.should()` is used in the tests that need it, instead of globally, in order to avoid a stack error in PhantomJS
- dropped SauceLabs and just use karma to orchestrate testing in local PhantomJS and Firefox

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    // level of logging
+    // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
+    logLevel: config.LOG_INFO,
+
+    // enable / disable watching file and executing tests whenever any file changes
+    autoWatch: true,
+
+    // start these browsers
+    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+    browsers: ['Firefox', 'PhantomJS'],
```
Comment (simonmarklar) [inline karma.conf.js]: I would switch this over to https://www.npmjs.com/package/karma-detect-browsers

**Comment by simonmarklar** [comment]
also, do you think  https://github.com/blinkmobile/bic-jqm/pull/20 should be pulled in first, its got a few tests in there that would need these updates ?

**Comment by jokeyrhyme** [comment]
Done

**Comment by jokeyrhyme** [comment]
I'm easy, merge #20 before or after, w/e.

**Pull request merged**